File: youqugou-interface/src/main/java/cn/itcast/core/service/TypeTemplateService.java
package cn.itcast.core.service;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.template.TypeTemplate;

import java.util.List;
import java.util.Map;

public interface TypeTemplateService {
    PageResult searchTypeTemplate(Integer pageNum, Integer pageSize, TypeTemplate typeTemplate);

    TypeTemplate selectTypeTemplateById(Long id);

    int updateTypeTemplate(TypeTemplate typeTemplate);

    int insertTypeTemplate(TypeTemplate typeTemplate);

    int deleteTypeTemplateByIds(Long[] ids);

    List<TypeTemplate> selectAllTypeTemplates();

    List<Map> selectSpecListById(Long id);

    List<Map> selectAllTypeTemplatesMap();
}

File: youqugou-web-manager/src/main/webapp/js/service/typeTemplateService.js
app.service("typeTemplateService", function ($http) {
    this.searchTypeTemplate = function (currentPage, itemsPerPage, searchEntity) {
        return $http.post("../typeTemplate/searchTypeTemplate.do?pageNum=" + currentPage
            + "&pageSize=" + itemsPerPage, searchEntity);
    }

    this.selectTypeTemplateById = function (id) {
        return $http.get("../typeTemplate/selectTypeTemplateById.do?id=" + id);
    }

    this.updateTypeTemplate = function (typeTemplate) {
        return $http.post("../typeTemplate/updateTypeTemplate.do", typeTemplate);
    }

    this.insertTypeTemplate = function (typeTemplate) {
        return $http.post("../typeTemplate/insertTypeTemplate.do", typeTemplate);
    }

    this.deleteTypeTemplateByIds = function (ids) {
        return $http.post("../typeTemplate/deleteTypeTemplateByIds.do", ids);
    }

    this.selectAllTypeTemplates = function () {
        return $http.get("../typeTemplate/selectAllTypeTemplates.do");
    }

    // 查询所有(id,name as text)用于select2
    this.selectAllTypeTemplatesMap = function () {
        return $http.get("../typeTemplate/selectAllTypeTemplatesMap.do");
    }
});